src/Operation/Chunk.php

```diff
@@ -18,47 +18,49 @@
 final class Chunk extends AbstractOperation
 {
     /**
-     * @return Closure(int...): Closure(iterable<TKey, T>): Generator<int, list<T>>
+     * @return Closure(int...): Closure(iterable<TKey, T>): Generator<int, non-empty-list<T>>
      */
     public function __invoke(): Closure
     {
         return
             /**
-             * @return Closure(iterable<TKey, T>): Generator<int, list<T>>
+             * @return Closure(iterable<TKey, T>): Generator<int, non-empty-list<T>>
              */
             static fn (int ...$sizes): Closure =>
                 /**
                  * @param iterable<TKey, T> $iterable
                  *
-                 * @return Generator<int, list<T>>
+                 * @return Generator<int, non-empty-list<T>>
                  */
                 static function (iterable $iterable) use ($sizes): Generator {
                     $sizesCount = count($sizes);
-                    $i = 0;
-                    $values = [];
+                    $chunkIndex = 0;
+                    $chunk = [];
```

Comment:
@drupol , I have slightly changed the variables names here, since it took me around 15 minutes to figure out what `$i`, `$values`, and `$sizes` variables mean and what the code below does. If you wish, I could revert it back to the previous version, but in my opinion the current implementation is better

Reply:
No no it's fine, it's actually a good idea.